### PR TITLE
Add gen_twiddles implementation in CUDA

### PR DESCRIPTION
**Pull request by matias-gonz**

### CODE REVIEW

**Review by MegaRedHand — APPROVED** [review]

File: gpu/src/cuda/shaders/twiddles.cu [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    //     const uint index = blockIdx.x * blockDim.x + threadIdx.x;
+
+    //     p256::Fp omega = _omega;
+    //     result[index] = omega.pow(index).inverse();
+    // };
+
+    __global__ void calc_twiddles_bitrev(p256::Fp *result,
+                                         const p256::Fp &_omega)
+    {
+        unsigned index = blockIdx.x * blockDim.x + threadIdx.x;
```
Comment (xqft) [inline gpu/src/cuda/shaders/twiddles.cu:29]:
```suggestion
        unsigned index = threadIdx.x;
```
idem

**Review by xqft — APPROVED** [review]
Nice work! I left some nits comments, also a unit test for `gen_twiddles()` would be great (now I realized Metal is missing it).

**Comment by matias-gonz** [comment]
Let's merge #268 first. Then we can merge this into main

File: gpu/src/cuda/shaders/utils.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+__device__ unsigned reverse_index(unsigned i, unsigned size)
+{
+    if (size == 1)
+    { // TODO: replace this statement with an alternative solution.
+        return i;
+    }
+    else
+    {
+        return __brev(i) >> (__clz(size) + 1);
+    }
```
Comment (Oppen) [inline gpu/src/cuda/shaders/utils.h:14]: If the branching is the issue, you can use the multiplication trick:
```suggestion
    return (size == 1) * i | (size != 1) * (__brev(i) >> (__clz(size) + 1));
```

**Review by Oppen — APPROVED** [review]

File: gpu/src/cuda/abstractions/state.rs [hunk 3]
```diff
@@ -90,10 +90,10 @@ impl CudaState {
...
 
         let (root, function_name) = match config {
-            RootsConfig::Natural => (root, "calc_twiddles"),
-            RootsConfig::NaturalInversed => (root.inv(), "calc_twiddles"),
-            RootsConfig::BitReverse => (root, "calc_twiddles_bitrev"),
-            RootsConfig::BitReverseInversed => (root.inv(), "calc_twiddles_bitrev"),
+            RootsConfig::Natural => (root, "calc_twiddless"),
+            RootsConfig::NaturalInversed => (root.inv(), "calc_twiddless"),
+            RootsConfig::BitReverse => (root, "calc_twiddles_bitrevs"),
+            RootsConfig::BitReverseInversed => (root.inv(), "calc_twiddles_bitrevs"),
```
Comment (Oppen) [inline gpu/src/cuda/abstractions/state.rs]: There's something odd with these names. The extra `s` at the end seems strange.

**Review by Oppen — APPROVED** [review]

**Pull request merged**